### ISSUE TITLE
Rate limit configuration sometimes fails

### ISSUE THREAD

**Issue by BenJChen** [getsentry/self-hosted#2079]
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Version

 23.2.0

### Steps to Reproduce

1.[Project] » Client Keys » Configure
2.Rate limit 5000 event(s) in 1hour
![image](https://user-images.githubusercontent.com/15866873/225192803-c37526dd-8faa-4d40-8e8d-c63fa28cc701.png)

### Expected Result

porject always rate limit 5000 event(s) in 1hour

### Actual Result

Sometimes the rate limit will fail, causing a large number of requests to enter and insufficient memory
![image](https://user-images.githubusercontent.com/15866873/225195622-1f226cff-12bc-42ec-b80d-14309c51768b.png)
![image](https://user-images.githubusercontent.com/15866873/225192905-cd1d405a-5693-43c6-a539-4dc674a13eff.png)

**Comment by hubertdeng123** [comment]
I'm wondering if because there are some many requests are entering, your system is getting overloaded. When you mention insufficient memory, are you saying that the RAM usage is high? Self-hosted Sentry isn't built for scale. Around how many events are coming in during a specific time frame?

**Comment by BenJChen** [comment]
> 

Yes, when there are the most requests, there are 20,000 to 30,000 requests per hour, which leads to Sentry failure. So we set a rate for projects that report too much. In version 22.90, the rate setting can effectively limit and keep sentry stable. When we upgrade to version 23.2, the rate limit seems to often fail and cause sentry to fail.

In addition, we also found that sentry has other errors, such as 
*** uWSGI listen queue of socket "127.0.0.1:41651" (fd: 3) full !!! (101/100) ***

Version 23.2 seems to be easier to service exceptions than version 22.9
